overlord/snapstate/snapstate.go

```diff
@@ -1354,7 +1354,52 @@ func canDisable(si *snap.Info) bool {
 	return true
 }
 
+// baseInUse returns true if the given base is needed by another snap
+func baseInUse(st *state.State, base *snap.Info) bool {
+	snapStates, err := All(st)
+	if err != nil {
+		return false
+	}
+	for _, snapst := range snapStates {
+		if !snapst.Active {
```

Comment:
The fact it's not active doesn't seem relevant for this case. How would we make a non-active snap active (snap enable, snap revert, etc) if its base is gone?